components/observatorium.libsonnet

```diff
@@ -207,7 +207,57 @@ local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
     ) +
     service.mixin.metadata.withNamespace(obs.config.namespace),
 
-  api:: (import 'observatorium/observatorium-api.libsonnet') + {
+  apiWithTLS:: (import 'observatorium/observatorium-api.libsonnet').withTLS {
+    config+:: {
+      tls: {
+        certFile: obs.config.apiWithTLS.certFile,
+        privateKeyFile: obs.config.apiWithTLS.privateKeyFile,
+        clientCAFile: obs.config.apiWithTLS.clientCAFile,
+        reloadInterval: obs.config.apiWithTLS.reloadInterval,
+      },
+    },
+    deployment+: {
+      spec+: {
+        template+: {
+          spec+: {
+            containers: [
+              if c.name == 'observatorium-api' then c {
+                volumeMounts+: [
+                  {
+                    name: obs.config.apiWithTLS.volumeMounts.name,
+                    mountPath: obs.config.apiWithTLS.volumeMounts.mountPath,
+                    readOnly: obs.config.apiWithTLS.volumeMounts.readOnly,
+                  },
+                ],
+                // Temporary workaround for the broken probes with this obs-api image
+                livenessProbe: {
+                  tcpSocket: {
+                    port: 8080,
+                  },
+                },
+                readinessProbe: {
+                  tcpSocket: {
+                    port: 8080,
+                  },
+                },
+              } else c
+              for c in super.containers
+            ],
+            volumes+: [
+              {
+                name: obs.config.apiWithTLS.volumeMounts.name,
+                secret: {
+                  secretName: obs.config.apiWithTLS.volumeMounts.secretName,
+                },
+              },
+            ],
+          },
+        },
+      },
+    },
+  },
+
+  api:: (import 'observatorium/observatorium-api.libsonnet') + obs.apiWithTLS {
```

Comment:
Shouldn't this be the basic component without TLS? TLS should be added with the mixin in an environment, eg base

Reply:
Done